proof/refine/ARM/Ipc_R.thy

```diff
@@ -5096,19 +5096,32 @@ crunches setThreadState
   (wp: crunch_wps)
 
 lemma valid_replies'_no_tcb_not_linked:
-  "\<lbrakk>obj_at' (\<lambda>ko. replyTCB ko = None) replyPtr s;
+  "\<lbrakk>replyTCBs_of s replyPtr = None;
     valid_replies' s; valid_replies'_sc_asrt replyPtr s\<rbrakk>
-    \<Longrightarrow> obj_at' (\<lambda>ko. replyPrev ko = None \<and> replyNext ko = None) replyPtr s"
+    \<Longrightarrow> \<not> is_reply_linked replyPtr s \<and> replySCs_of s replyPtr = None"
   apply (clarsimp simp: valid_replies'_def valid_replies'_sc_asrt_def)
   apply (drule_tac x=replyPtr in spec)
-  apply (clarsimp simp: obj_at'_real_def ko_wp_at'_def opt_map_def projectKO_reply)
-  apply (case_tac "replyNext obj"; simp)
-  apply (case_tac a; simp)
+  apply (clarsimp simp: obj_at'_real_def ko_wp_at'_def opt_map_def
+                        projectKO_reply)
   done
 
+lemma obj_at'_SCReply_ref:
+  "obj_at' (\<lambda>a. scReply a = Some rp) scp s
+   \<Longrightarrow> (rp, SCReply) \<in> state_refs_of' s scp"
+   by (clarsimp simp: obj_at'_real_def ko_wp_at'_def state_refs_of'_def
+                      projectKO_sc get_refs_def)
+
+lemma ReplySchedContext_obj_at'_ref:
+  "(scp, ReplySchedContext) \<in> state_refs_of' s rptr
+   \<Longrightarrow> reply_at' rptr s
+   \<Longrightarrow> replySCs_of s rptr = Some scp"
+   by (clarsimp simp: obj_at'_real_def ko_wp_at'_def state_refs_of'_def
+                      projectKOs get_refs_def opt_map_def
+               split: option.splits)
+
```

Comment:
I removed these in my clean up PR that was merged last week. I believe these are not used anymore. Maybe something has gone a bit funny while rebasing.